### PR TITLE
Split supplier module into 2 modules: "Purchase order" and "Supplier invoice" module (to match customer side)

### PR DESCRIPTION
**Pull request by aspangaro**

### CODE REVIEW

File: htdocs/comm/action/class/cactioncomm.class.php [hunk 1]
```diff
@@ -169,8 +169,8 @@ public function liste_array($active = '', $idorcode = 'id', $excludetype = '', $
                         if ($obj->module == 'invoice' && ! $conf->facture->enabled)	 $qualified=0;
                         if ($obj->module == 'order'   && ! $conf->commande->enabled) $qualified=0;
                         if ($obj->module == 'propal'  && ! $conf->propal->enabled)	 $qualified=0;
-                        if ($obj->module == 'invoice_supplier' && ! $conf->fournisseur->enabled)   $qualified=0;
-                        if ($obj->module == 'order_supplier'   && ! $conf->fournisseur->enabled)   $qualified=0;
+                        if ($obj->module == 'invoice_supplier' && (! $conf->fournisseur->enabled && $conf->global->MAIN_USE_NEW_SUPPLIERMOD || ! $conf->supplier_invoice->enabled))	$qualified=0;
```
Comment (eldy) [inline htdocs/comm/action/class/cactioncomm.class.php]: a & b | c
Some parenthesis are missing surely.

Comment (aspangaro) [reply htdocs/comm/action/class/cactioncomm.class.php]: It's ok now @eldy ?

File: htdocs/comm/action/class/cactioncomm.class.php [hunk 2]
```diff
@@ -169,8 +169,8 @@ public function liste_array($active = '', $idorcode = 'id', $excludetype = '', $
                         if ($obj->module == 'invoice' && ! $conf->facture->enabled)	 $qualified=0;
                         if ($obj->module == 'order'   && ! $conf->commande->enabled) $qualified=0;
                         if ($obj->module == 'propal'  && ! $conf->propal->enabled)	 $qualified=0;
-                        if ($obj->module == 'invoice_supplier' && ! $conf->fournisseur->enabled)   $qualified=0;
-                        if ($obj->module == 'order_supplier'   && ! $conf->fournisseur->enabled)   $qualified=0;
+                        if ($obj->module == 'invoice_supplier' && ((! $conf->fournisseur->enabled && $conf->global->MAIN_USE_NEW_SUPPLIERMOD) || ! $conf->supplier_invoice->enabled))	$qualified=0;
```
Comment (eldy) [inline htdocs/comm/action/class/cactioncomm.class.php]: Are you sure this is not instead 
if ($obj->module == 'invoice_supplier' && ((! $conf->fournisseur->enabled && empty($conf->global->MAIN_USE_NEW_SUPPLIERMOD)) || ! $conf->supplier_invoice->enabled))	$qualified=0;

Comment (aspangaro) [reply htdocs/comm/action/class/cactioncomm.class.php]: You right, it's fix now.

File: htdocs/admin/workflow.php [hunk 3]
```diff
@@ -85,11 +85,11 @@
 	'WORKFLOW_INVOICE_AMOUNT_CLASSIFY_BILLED_ORDER'=>array('family'=>'classify_order', 'position'=>41, 'enabled'=>'! empty($conf->facture->enabled) && ! empty($conf->commande->enabled)', 'picto'=>'order', 'warning'=>''), // For this option, if module invoice is disabled, it does not exists, so "Classify billed" for order must be done manually from order card.
     'separator2'=>array('family'=>'separator', 'position'=>50),
 	// Automatic classification supplier proposal
-	'WORKFLOW_ORDER_CLASSIFY_BILLED_SUPPLIER_PROPOSAL'=>array('family'=>'classify_supplier_proposal', 'position'=>60, 'enabled'=>'! empty($conf->supplier_proposal->enabled) && ! empty($conf->fournisseur->enabled)', 'picto'=>'propal', 'warning'=>''),
+	'WORKFLOW_ORDER_CLASSIFY_BILLED_SUPPLIER_PROPOSAL'=>array('family'=>'classify_supplier_proposal', 'position'=>60, 'enabled'=>'! empty($conf->supplier_proposal->enabled) && (!empty($conf->fournisseur->enabled) && empty($conf->global->MAIN_USE_NEW_SUPPLIERMOD) || !empty($conf->supplier_order->enabled) || !empty($conf->supplier_invoice->enabled))', 'picto'=>'propal', 'warning'=>''),
```
Comment (eldy) [inline htdocs/admin/workflow.php]: Still pb with parenthesis

**Comment by altatof** [comment]
Hello @aspangaro and @eldy ; it 's typically the type of PR that seems to do big stuff in dolibarr but have no description, so it is impossible to understand what we are talking about and so impossible to debate.
I think PR description should be mandatory or at least that PR should be refused if the goal of the PR is not obvious.
For instance, here, are we talking about splitting the supplier module into several modules like on the customer side (proposals, orders, invoices, ...) ? or is it something else ?
I took a look at the code quickly but only saw a new MAIN_USE_NEW_SUPPLIERMOD option... what are we talking about exactly here ?

**Comment by altatof** [comment]
more than this, i think that big developments should be relative to an issue created and debated before PRs.

**Pull request merged**